### ISSUE TITLE
version 3.3.5 build fails with build tree separate from source

### ISSUE THREAD

**Issue by matzeri** [FFTW/fftw3#66]
to replicate:
```
cd <source_tree>
autoreconf -ifv
cd <build_tree>
<source_tree>/configure  \
                --enable-maintainer-mode \
                --enable-float \
                --enable-threads \
                --enable-shared --disable-static \
                --enable-openmp \
                --with-windows-f77-mangling

```
The outcome:

```
ocamlbuild -classic-display -libs unix,nums gen_notw.native gen_notw_c.native gen_twiddle.native gen_twiddle_c.native gen_twidsq.native gen_twidsq_c.native gen_r2r.native gen_r2cf.native gen_r2cb.native gen_hc2c.native gen_hc2cdft.native gen_hc2cdft_c.native gen_hc2hc.native gen_mdct.native
Solver failed:
  Ocamlbuild cannot find or build gen_notw.ml.  A file with such a name would usually be a source file.  I suspect you have given a wrong target name to Ocamlbuild.
Backtrace:
  - Failed to build the target gen_notw.native
```
where `genfft/gen_notw.ml` is on source tree.

Without using `enable-maintainer-mode` the build fails like https://github.com/FFTW/fftw3/issues/56

Additional comment: as there is a announce mailing list but not a devel list, it will be better to announce also the release candidate to receive prompt feedback on them before release.

**Comment by matteo-frigo** [comment]
Separate build tree for --enable-maintainer-mode was never meant to work and has never worked in 20 years.  The whole point of --enable-maintainer-mode (in fftw) is to generate the source code in the source tree, so there was never any desire to make it work out of tree.

The intent is that the maintainer uses --enable-maintainer-mode in the source tree, runs "make dist" to produce a tarball, and the tarball can be compiled in a build tree without requiring ocaml.  Users download fftw-3.3.5.tar.gz and don't have to know how the sausage was made.

Thus, my position on this issue is WONTFIX---let me know if you disagree.

**Comment by matzeri** [comment]
I guess your `make dist` is not working correctly on 3.5.0.
The build fails without `enable-maintainer-mode` , and this is a regression
from previous 3.x.. versions. 
The following method always worked, as I used it to build cygwin packages:

```
cd <source_tree>
autoreconf -ifv
cd <build_tree>
<source_tree>/configure  \
                --enable-float \
                --enable-threads \
                --enable-shared --disable-static \
                --enable-openmp \
                --with-windows-f77-mangling

[cut]
make[4]: Entering directory '/cygdrive/e/cyg_pub/devel/fftw/prova2/fftw3-3.3.5-1.x86_64/build/float/dft/scalar/codelets'
make[4]: *** No rule to make target 'n1_2.c', needed by 'all'.  Stop.
```

As this failed I tried to use the `enable-maintainer-mode` but also that failed. 
My scope is only to build on separate trees. I have no particular interest on  `enable-maintainer-mode`

**Comment by matteo-frigo** [comment]
I think we are talking about two different things and I am not sure I understand what you are saying.

There are two cases: either you download fftw-3.3.5.tar.gz from fftw.org, or you checkout from github.

If you start from fftw-3.3.5.tar.gz, you are *not* supposed to use --enable-maintainer-mode, you are not supposed to run autoreconf (although it might work), and your configure command is supposed to work from a separate build tree.  (I have tried exactly the same procedure in the past couple of days on all of debian lenny/squeeze/wheezy/jessie/stretch and centos 5/6/7, plus various ubuntu versions and openbsd, so I expect it to work by now.)

If you start from github, what you are trying to do won't work, and it didn't work in 3.3.4 either (I just tried).  To start from github you have to run the bootstrap.sh script in the source directory and run "make dist" at least once in the source directory.  This will generate all source files, n1_2.c being the first one.  After that, you are in the same situation as if you had downloaded fftw-3.3.5.tar.gz from fftw.org and you can proceed as in the first case.

This may seem complicated and it is, but our intent is that we distribute tarballs to non-maintainers, so that we and users don't have to worry about whether ocaml works on AIX or Android. 

I guess the difference is that we put stuff on github only after 3.3.4, so you must have started from the tarball for 3.3.4 and you are starting from github now.

**Comment by matzeri** [comment]
Can you please put a note on 
https://github.com/FFTW/fftw3/releases 
to advise that the release files are incomplete and to use instead
http://www.fftw.org/download.html 

I missed the difference between the two versions.
The second one works fine

**Comment by matteo-frigo** [comment]
> Can you please put a note on https://github.com/FFTW/fftw3/releases

WTF?  I didn't realize that github was creating releases automatically without asking me.  Do you know how to disable them?  The only releases we ever meant to publish are at fftw.org.

**Comment by matzeri** [comment]
Sorry no clue. 
I guess https://github.com/FFTW/fftw3/issues/56 is the same problem

**Comment by matteo-frigo** [comment]
I have added a few warnings in README.md and the github release page.  Let me know if you can think of something else I could do to avoid confusing people.
